src/main/java/macrobase/analysis/result/DatumWithNorm.java

```diff
@@ -0,0 +1,26 @@
+package macrobase.analysis.result;
+
+import macrobase.datamodel.Datum;
+
+public class DatumWithNorm {
+    private final Datum datum;
+    private final double norm;
+
+    public Double getNorm() {
+        return norm;
+    }
+
+    public Datum getDatum() {
+        return datum;
+    }
+
+    public DatumWithNorm(Datum d) {
+        this.datum = d;
+        this.norm = d.getMetrics().getNorm();
```

Comment:
What's the point of DatumWithNorm if the norm is already available in a standard Datum?